carla/models/catalog/catalog.py

```diff
@@ -0,0 +1,154 @@
+                "relationship_Non-Husband",
+                "race_White",
+                "sex_Male",
+                "native-country_US",
+            ]
+        else:
+            raise Exception("Model for dataset not in catalog")
+
+    @property
+    def name(self):
+        """
+        Contains meta information about the model, like name, dataset, structure, etc.
+
+        Returns
+        -------
+        name : String
+            Individual name of the ml model
+        """
+        return self._name
+
+    @property
+    def feature_input_order(self):
+        """
+        Saves the required order of feature as list.
+
+        Prevents confusion about correct order of input features in evaluation
+
+        Returns
+        -------
+        ordered_features : List of String
+            Correct order of input features for ml model
+        """
+        return self._feature_input_order
+
+    @property
+    def backend(self):
+        """
+        Describes the type of backend which is used for the ml model.
+
+        E.g., tensorflow, pytorch, sklearn, ...
+
+        Returns
+        -------
+        backend : String
+            Used framework
+        """
+        return self._backend
+
+    @property
+    def raw_model(self):
+        """
+        Returns the raw ml model built on its framework
+
+        Returns
+        -------
+        ml_model : tensorflow, pytorch, sklearn model type
+            Loaded model
+        """
+        return self._model
+
+    def predict(self, x):
+        """
+        One-dimensional prediction of ml model for an output interval of [0, 1]
+
+        Shape of input dimension has to be always two-dimensional (e.g., (1, m), (n, m))
+
+        Parameters
+        ----------
+        x : np.Array or pd.DataFrame
+            Tabular data of shape N x M (N number of instances, M number of features)
+
+        Returns
+        -------
+        output : np.Array
+            Ml model prediction for interval [0, 1] with shape N x 1
+        """
+
+        assert len(x.shape) == 2
+
+        if self._backend == "pytorch":
+            output = self._model.predict(x)
+        elif self._backend == "tensorflow":
+            output = self._model.predict(x)[:, 1]
+
+        return output
+
+    def predict_proba(self, x):
+        """
+        Two-dimensional softmax prediction of ml model
+
+        Shape of input dimension has to be always two-dimensional (e.g., (1, m), (n, m))
+
+        Parameters
+        ----------
+        x : np.Array or pd.DataFrame
+            Tabular data of shape N x M (N number of instances, M number of features)
+
+        Returns
+        -------
+        output : float
+            Ml model prediction with shape N x 2
+        """
+
+        assert len(x.shape) == 2
+
+        if self._backend == "pytorch":
+            class_1 = 1 - self._model.forward(x).detach().numpy().squeeze()
+            class_2 = self._model.forward(x).detach().numpy().squeeze()
```

Comment:
There is no easier way for pytorch models to receive the probability? 🙈

Reply:
Not as far as I know, the output layer in this structure is one-dimensional.
Since we have problems with loading the current pre-trained pytorch models, we could avoid this by changing the model architecture if we have to retrain these.

Reply:
I was more thinking something along the lines of tensorflow where its particularly easey `self._model.predict(x)`, but appearently not so much with pytorch :(